primitives/state-machine/src/lib.rs

```diff
@@ -68,11 +79,12 @@ macro_rules! warn {
 #[cfg(not(feature = "std"))]
 #[macro_export]
 macro_rules! debug {
-	(target: $target:expr, $($arg:tt)+) => {
-		()
-	};
-	($($arg:tt)+) => {
-		()
+	(target: $target:expr, $message:expr, $( $arg:ident ),*) => {
```

Comment:
same maybe if my suggestion works we can do same as `warn`
```suggestion
	(target: $target:expr, $message:expr $( , $arg:ident ) * $(,)? ) => {
```